### PR TITLE
Pulled the getContacts stuff out into its own class and got the email and phone number fields returning on older versions of android

### PR DESCRIPTION
**Pull request by npomfret**
Pulled the getContacts stuff out into its own class and got the email and phone number fields returning on older versions of android

### CODE REVIEW

File: android/src/main/java/com/rt2zz/reactnativecontacts/ContactsProvider.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        WritableArray contacts = Arguments.createArray();
+        for (Contact contact : contactList) {
+            contacts.pushMap(contact.toMap());
+        }
+
+        return contacts;
+    }
+
+    private Contact createContact(Cursor cursor) {
+        int contactId = cursor.getInt(cursor.getColumnIndex(CONTACT_ID));
```
Comment (pickhardt) [inline android/src/main/java/com/rt2zz/reactnativecontacts/ContactsProvider.java]: I've done some research, and you can optimize the performance by getting the column indices only once, before the while loop. This way you're not looking it up for every contact.

**Comment by pickhardt** [comment]
Maybe use runQueryOnBackgroundThread? It's basically crashing my phone when I connect contacts, unfortunately, because it takes a long time.

**Comment by npomfret** [comment]
Thanks for testing, I wasn't sure how bad multiple cursors would be.  Here's another solution with a single cursor https://github.com/rt2zz/react-native-contacts/pull/49

**Comment by npomfret** [comment]
I've got the structured name stuff working too. Should have included it in my pr but forgot, I'll send it through if people are happy with #49 

https://github.com/npomfret/react-native-contacts/blob/master/android/src/main/java/com/rt2zz/reactnativecontacts/ContactsProvider.java

**Pull request closed without merging**